text/0000-compiler-requirements.md

```diff
@@ -0,0 +1,92 @@
+
+### Current Tools
+
+fontmake. It's the glue that binds together ufo2ft, fontTools and some other codebases to take any of the usual font formats (UFOs, Designspace + UFOs and Glyphs.app files) and produce font binaries (TTF, CFF(2); static, interpolated static, variable).
+
+### Python Compiler Pipeline Libraries
+
+The Python library support requirements for font compilation include:
+
+- [fonttools](): I/O, lexing/parsing, intermediate code generation, optimization, code generation
+- [cu2qu](): intermediate code optimization
+- [glyphsLib](): I/O, lexing/parsing
+- [ufo2ft](): 
+- [MutatorMath](): linear interpolation calculation
+- [fontMath]()
+- [defcon]()
+- [booleanOperations]()
+- [ufoLib2]()
+- [attrs]()
+- [cffsubr]()
+- [compreffor]()
+- [ttfautohint-py]()
+
+### Object Code Specifications
+
+- otf
+- otf-cff2
+- ttf
+- ttf-interpolatable
+- otf-interpolatable
+- OpenType variable font ttf format
+- OpenType variable font cff2 format
+
+### Key Limitations
+
+- Python is bad at embarrasingly parallel tasks like instance generation and glyph compilation due to overhead and lacking multicore facilities.
+- The memory consumption of the compilation process can hinder the amount of build processes running in parallel.
+- It's easy to run into performance problems with OpenType Layout table repacking, which quickly consumes a large part of the compile time.
+- Codebases have grown organically by adding new things on top, some parts are byzantine and hard to change, making optimizations or rearranging code paths hard.
+  - VF generation requires compiling all masters fully and then merging them, instead of generating variable data directly
+  - fontmake is underloved from a maintenance perspective, with some low hanging performance fruits continuing to hang.
+  - The layering of fontmake → ufo2ft → fontTools makes changing the architecture hard. User-relevant options need to be made available and passed through all layers, e.g. GPOS compression recently
+  - Compilation of e.g. Glyphs.app files requires converting them into UFOs because ufo2ft can only compile those, which runs into impedance mismatches and is highly fragile and error-prone.
+
+## What Do We Need?
+
+An at least 10x faster compiler using less memory.
+
+### Must Have
+
+### Good to Have
+
+### Someday
+
+### Out of Scope
+
+## Why Do We Need It?
+
+Projects like Noto take 30+ minutes to compile completely, and a certain secret font project takes 5+ minutes to compile even with Pypy. Given how fickle font sources are and the experimentation and scripting needed to achieve certain design goals, this severly hinders incremental improvements during font design and engineering. Additionally, the resource waste is a problem for paid-for continuous integration.
+
+## When Do We Need It?
+
+As soon as possible.
+
+## Implementation Requirements
```

Comment:
At a glance none of these appear to be **requirements**. Recommendations might be more appropriate.